src/algebra/ring/defs.lean

```diff
@@ -292,9 +292,9 @@ class non_unital_ring (α : Type*) extends
   non_unital_non_assoc_ring α, non_unital_semiring α
 
 /-- A unital but not-necessarily-associative ring. -/
-@[protect_proj, ancestor non_unital_non_assoc_ring non_assoc_semiring]
+@[protect_proj, ancestor non_unital_non_assoc_ring non_assoc_semiring add_comm_group_with_one]
 class non_assoc_ring (α : Type*) extends
-  non_unital_non_assoc_ring α, non_assoc_semiring α, add_group_with_one α
+  non_unital_non_assoc_ring α, non_assoc_semiring α, add_comm_group_with_one α
```

Comment:
Can you make a forward-port of just this line and check it doesn't cause any TC problems in CI? The rest of the forward-port can be committed later.

Reply:
I just checked and `NonAssocRing` already extends `AddCommGroupWithOne`.